go/arrow/avro/schema.go

```diff
@@ -0,0 +1,409 @@
+	case "date":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Date32}
+
+	// The time-millis logical type represents a time of day, with no reference to a particular calendar,
+	// time zone or date, with a precision of one millisecond.
+	// A time-millis logical type annotates an Avro int, where the int stores the number of milliseconds
+	// after midnight, 00:00:00.000.
+	case "time-millis":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Time32ms}
+
+	// The time-micros logical type represents a time of day, with no reference to a particular calendar,
+	// time zone or date, with a precision of one microsecond.
+	// A time-micros logical type annotates an Avro long, where the long stores the number of microseconds
+	// after midnight, 00:00:00.000000.
+	case "time-micros":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Time64us}
+
+	// The timestamp-millis logical type represents an instant on the global timeline, independent of a
+	// particular time zone or calendar, with a precision of one millisecond. Please note that time zone
+	// information gets lost in this process. Upon reading a value back, we can only reconstruct the instant,
+	// but not the original representation. In practice, such timestamps are typically displayed to users in
+	// their local time zones, therefore they may be displayed differently depending on the execution environment.
+	// A timestamp-millis logical type annotates an Avro long, where the long stores the number of milliseconds
+	// from the unix epoch, 1 January 1970 00:00:00.000 UTC.
+	case "timestamp-millis":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Timestamp_ms}
+	// The timestamp-micros logical type represents an instant on the global timeline, independent of a
+	// particular time zone or calendar, with a precision of one microsecond. Please note that time zone
+	// information gets lost in this process. Upon reading a value back, we can only reconstruct the instant,
+	// but not the original representation. In practice, such timestamps are typically displayed to users
+	// in their local time zones, therefore they may be displayed differently depending on the execution environment.
+	// A timestamp-micros logical type annotates an Avro long, where the long stores the number of microseconds
+	// from the unix epoch, 1 January 1970 00:00:00.000000 UTC.
+	case "timestamp-micros":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Timestamp_us}
+	// The local-timestamp-millis logical type represents a timestamp in a local timezone, regardless of
+	// what specific time zone is considered local, with a precision of one millisecond.
+	// A local-timestamp-millis logical type annotates an Avro long, where the long stores the number of
+	// milliseconds, from 1 January 1970 00:00:00.000.
+	case "local-timestamp-millis":
+		return arrow.Field{Name: node.name, Type: arrow.FixedWidthTypes.Timestamp_ms}
```

Comment:
the default `arrow.FixedWidthTypes.Timestamp_ms` sets the timezone to "UTC". You should manually create `&arrow.TimestampType{Unit: arrow.Millisecond}` for the local timestamp (or maybe we should create `NoTZ` versions of these vars that you can use for this....?

Reply:
perhaps NoTZ versions would be better - I was going on the assumption that the users of the data would have to cast the timezone to the data.

Reply:
implemented as `&arrow.TimestampType{Unit: arrow.Millisecond}`